### DISCUSSION TITLE
Questions on cgns grid file; which is to be read and solved using ADFLOW

### DISCUSSION THREAD

**Discussion by vaths1** [mdolab/MACH-Aero#148] [Q&A]
Hello, 
I am new to working with cgns grid files; I would be asking couple of dumb or obvious questions, I really hope you can still answer them.

Q1. When I am generating a **2D/3D  structured grid using pointwise** and read it into the ADflow runscript, what would be the right solver to be selected?
I have chosen the cgns obviously, since MACH-AERO was not listed, I just want to make sure it's the right place to get started.
<img width="347" alt="image" src="https://github.com/mdolab/MACH-Aero/assets/151889110/ca7cf2a0-7178-4a8e-808f-be1b31fa41e6">

Q2 a) When I generate grid in Pointwise (for 2D), will a 2D surface mesh alone be sufficient (a single domain) or should I extrude it to a volume mesh (a block) _within pointwise itself_; just like how the airfoil mesh is generated in the tutorial. (https://mdolab-mach-aero.readthedocs-hosted.com/en/latest/machAeroTutorials/airfoilopt_mesh.html).

b) In continuation to first part of Q2a; or would it be prefereable to read the surface mesh into _pyhyp and then generate a volume_ instead? which would likely help with reading grid file into the solver...

Q3. I am trying to solve for an 2D external flow of a mini-turbojet engine, as shown in the image below; the domain and grid. 
CFD Domain:

![mail](https://github.com/mdolab/MACH-Aero/assets/151889110/f5e0ebf3-6f1a-432b-92f3-c92421a21d1b)

Grid: 

<img width="671" alt="image" src="https://github.com/mdolab/MACH-Aero/assets/151889110/a5564fe9-7451-410b-b12f-7d8eb71a737a">

**When I try to read the cgns file into the solver** I get the following error; seems like solver cannot read the grid file. :/
<img width="544" alt="image" src="https://github.com/mdolab/MACH-Aero/assets/151889110/d7532559-fb80-40d5-84d0-0ee318f6464d">
I strognly feel it has to do with the grid file format or data inside it which are not favorable to be opened and read. I am not able to get past this.!  (However when I read the same grid file into Paraview it reads perefectly fine just like the other airfoil grids generated from the tutorial.)

I would really appreciate asistance with this issue!

**Comment by gawng** [comment]
The tutorial on overset meshing should answer all of your questions: https://mdolab-mach-aero.readthedocs-hosted.com/en/latest/machAeroTutorials/overset_surface_mesh.html

But in short,
Q1: Yes, CGNS 2D
Q2: Check overset tutorial
Q3: Double check your file paths maybe? Normally if there is a data format issue, a warning is printed out by ADflow saying so.
